jdaviz/configs/default/plugins/markers/markers.py

```diff
@@ -107,15 +113,60 @@ def clear_table_callback():
         self.hub.subscribe(self, RemoveDataMessage,
                            handler=lambda msg: self._recompute_mark_positions(msg.viewer))
 
-        self.docs_description = "Press 'm' with the cursor over a viewer to log\
+        self.docs_description = """Press 'm' with the cursor over a viewer to log\
                                 the mouseover information. To change the\
                                 selected layer, click the layer cycler in the\
                                 mouseover information section of the app-level\
-                                toolbar."
+                                toolbar.
+Press 'd' twice to measure the distance between any two points. Hold the 'Option'
+(Mac) or 'Alt' key while pressing 'd' to snap measurement points to the nearest
+marker."""
 
         # description displayed under plugin title in tray
         self._plugin_description = 'Create markers on viewers.'
 
+    @traitlets.default('distances_table')
+    def _default_distances_table(self):
+        table = Table(
+            self,
+            title='Measurements',
+            headers=['Start RA', 'Start Dec', 'End RA', 'End Dec',
+                     'Separation (arcsec)', 'Distance (pix)', 'Position Angle (deg)'],
+        )
+        table.items_per_page = 15
+        table.hide_items_per_page = True
+        return table
+
+    def _clear_all_data(self):
+        """
+        Clears all data from the tables AND all marks from the viewer.
+        This is called by the "Clear Table" UI button.
+        """
+        self.table.clear_table()
+        if self.distances_table is not None:
+            self.distances_table.clear_table()
+        self.clear_viewer_marks()
+
+    def clear_viewer_marks(self):
+        """
+        Clears all visual marks from all viewers without clearing the data tables.
+        This is called by the 'r' key press.
+        """

```

Comment:
I would vote that we keep the previous behavior of two-way sync between the tables and displayed marks (and not allow clearing marks separate of the table)